mapic/misttriggers/stream_buffer.go

```diff
@@ -0,0 +1,192 @@
+	SessionID  string                  `json:"session_id"`
+	IsActive   bool                    `json:"is_active"`
+	IsHealthy  bool                    `json:"is_healthy"`
+	Tracks     map[string]TrackDetails `json:"tracks,omitempty"`
+	Issues     string                  `json:"issues,omitempty"`
+}
+
+func PostStreamHealthPayload(url, apiToken string, payload StreamHealthPayload) error {
+	body, err := json.Marshal(payload)
+	if err != nil {
+		return fmt.Errorf("error marshalling stream health payload: %w", err)
+	}
+	req, err := http.NewRequest("POST", url, bytes.NewReader(body))
+	if err != nil {
+		return fmt.Errorf("error creating stream health request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+apiToken)
+
+	resp, err := hookClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("error pushing stream health to hook: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 400 {
+		respBody, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("error reading stream health hook response: %w", err)
+		}
+		glog.Warningf("Error pushing stream health to hook status=%d body=%q", resp.StatusCode, respBody)
+	}
+
+	return nil
+}
+
+type StreamBufferPayload struct {
+	StreamName string
+	State      string
+	Details    *MistStreamDetails
+}
+
+type TrackDetails struct {
+	Codec  string                 `json:"codec"`
+	Kbits  int                    `json:"kbits"`
+	Keys   map[string]interface{} `json:"keys"`
+	Fpks   int                    `json:"fpks,omitempty"`
+	Height int                    `json:"height,omitempty"`
+	Width  int                    `json:"width,omitempty"`
+}
+
+func ParseStreamBufferPayload(lines []string) (*StreamBufferPayload, error) {
+	if len(lines) < 2 || len(lines) > 3 {
+		return nil, fmt.Errorf("invalid payload: expected 2 or 3 lines but got %d", len(lines))
+	}
+
+	streamName := lines[0]
+	streamState := lines[1]
+	var streamDetailsStr string
+	if len(lines) == 3 {
+		streamDetailsStr = lines[2]
+	}
+
+	streamDetails, err := ParseMistStreamDetails(streamState, []byte(streamDetailsStr))
+	if err != nil {
+		return nil, fmt.Errorf("error parsing stream details JSON: %w", err)
+	}
+
+	return &StreamBufferPayload{
+		StreamName: streamName,
+		State:      streamState,
+		Details:    streamDetails,
+	}, nil
+}
+
+type MistStreamDetails struct {
+	Tracks map[string]TrackDetails
+	Issues string
+}
+
+// Mists saves the tracks and issues in the same JSON object, so we need to
+// parse them separately. e.g. {track-id-1: {...}, issues: "..."}
+func ParseMistStreamDetails(streamState string, data []byte) (*MistStreamDetails, error) {
+	if streamState == "EMPTY" {
+		return nil, nil
+	}
+
+	var tracksAndIssues map[string]interface{}
+	err := json.Unmarshal(data, &tracksAndIssues)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing stream details JSON: %w", err)
+	}
+
+	issues, ok := tracksAndIssues["issues"].(string)
```

Comment:
I'm wondering if we need to check the key exists first before checking type, does this not panic with a nil pointer?

Reply:
I had that doubt myself, but it does work! Map fetches will never panic, they just return the zero value for the type, while interface casts work on completely `nil` interfaces _as long as you use the 2 value ok syntax_. 

For curiosity: The only caveat is when the `interface{}` is not `nil` but holds a nullable-type and a `nil` value. Then you could need to check if the value itself is also not nil additionally from the `ok` (and you can't use single-value syntax or that would explode)

playground: https://go.dev/play/p/WcZ_u4puwfW